File: data/EXP01/go/2502/2502.output_1.go
type Allocator struct {
    memory []int
    allocatedBlocks map[int][]int
}

func Constructor(n int) Allocator {
    memory := make([]int, n)
    allocatedBlocks := make(map[int][]int)
    
    return Allocator{
        memory: memory,
        allocatedBlocks: allocatedBlocks,
    }
}

func (this *Allocator) Allocate(size int, mID int) int {
    for i := 0; i <= len(this.memory)-size; i++ {
        if this.memory[i] == 0 {
            isFreeBlock := true
            for j := i; j < i+size; j++ {
                if this.memory[j] != 0 {
                    isFreeBlock = false
                    break
                }
            }
            if isFreeBlock {
                for j := i; j < i+size; j++ {
                    this.memory[j] = mID
                }
                this.allocatedBlocks[mID] = append(this.allocatedBlocks[mID], i)
                return i
            }
        }
    }
    return -1
}

func (this *Allocator) Free(mID int) int {
    count := 0
    if blocks, ok := this.allocatedBlocks[mID]; ok {
        for _, block := range blocks {
            for i := block; i < block+len(blocks); i++ {
                this.memory[i] = 0
            }
            count += len(blocks)
        }
        delete(this.allocatedBlocks, mID)
    }
    return count
}

File: data/EXP01/python/2502/2502.output_1.py
# @lc app=leetcode id=2502 lang=python3
class Allocator:

    def __init__(self, n: int):
        self.memory = [0] * n  # initialize memory array with all units free

    def allocate(self, size: int, mID: int) -> int:
        count = 0
        for i in range(len(self.memory)):
            if self.memory[i] == 0:
                count += 1
                if count == size:
                    for j in range(i - size + 1, i + 1):
                        self.memory[j] = mID
                    return i - size + 1
            else:
                count = 0
        return -1

    def free(self, mID: int) -> int:
        count = 0
        for i in range(len(self.memory)):
            if self.memory[i] == mID:
                count += 1
                self.memory[i] = 0
            else:
                if count > 0:
                    return count
                count = 0
        return count